lib/mongoid/criteria/scrollable.rb

```diff
@@ -68,7 +78,7 @@ def cursor_from_record(record, cursor_options)
       def scroll_field_type(criteria)
         scroll_field = scroll_field(criteria)
         field = criteria.klass.fields[scroll_field.to_s]
-        field.foreign_key? && field.object_id_field? ? bson_type : field.type
+        field.foreign_key? && field.object_id_field? ? bson_type.to_s : field.type.to_s
```

Comment:
It shouldn't create issues because `to_s` is already called on it in [`Mongoid::Scroll::Cursor.extract_field_options`](https://github.com/mongoid/mongoid-scroll/blob/a7993bb81ed0ad2faf62e9fa4792bf5e25d76d1e/lib/mongoid/scroll/cursor.rb#L61). But it's needed to compare the different options, otherwise we have a mismatch between `BSON::ObjectId` and `"BSON::ObjectId"`.